src/llmcompressor/transformers/sparsification/sparse_model.py

```diff
@@ -1,174 +1,29 @@
 
 import torch
-from accelerate import load_checkpoint_and_dispatch
-from compressed_tensors.compressors import ModelCompressor
-from compressed_tensors.quantization import (
-    QuantizationStatus,
-    apply_quantization_config,
-)
 from loguru import logger
 from torch.nn import Module
-from transformers import AutoModelForCausalLM, PreTrainedModel
+from transformers import AutoModelForCausalLM
 
 from llmcompressor.pytorch.model_load.helpers import initialize_recipe
-from llmcompressor.transformers.sparsification.compressed_tensors_utils import (
-    modify_save_pretrained,
-    patch_tied_tensors_bug,
-)
-from llmcompressor.transformers.utils.helpers import (
-    download_model_directory,
-    resolve_recipe,
-)
+from llmcompressor.transformers.utils.helpers import resolve_recipe
 
 __all__ = [
-    "wrap_hf_model_class",
     "SparseAutoModel",
     "SparseAutoModelForCausalLM",
     "get_shared_tokenizer_src",
 ]
 
 
-def wrap_hf_model_class(hf_model_class: PreTrainedModel) -> PreTrainedModel:
-    """
-    Wrap a HF PreTrainedModel class to
-    1. Decompress a compressed model
-    2. Initialize any saved recipes
-    3. Wrap the `save_pretrained` method to allow saving as a compressed model
-
-    :param hf_model_class: Model class to wrap
-    :return: Wrapped model class
-    """
-
-    # Add the from_pretrained class method
-    @classmethod
-    def from_pretrained(
-        cls,
-        pretrained_model_name_or_path,
-        run_compressed: bool = False,
-        recipe: Optional[Union[str, Path]] = None,
-        *model_args,
-        **kwargs,
-    ) -> PreTrainedModel:
-        """
-        A wrapper around the PreTrainedModel.from_pretrained method
-
-        :param pretrained_model_name_or_path: the name of or path to the model to load
-        :param recipe: the path to the recipe file to apply to the model. Can be a
-            string or Path object. If None, a recipe will be searched for in the
-            pretrained_model_name_or_path directory and applied if found
-        :return the created model for causal language modeling
-        """
-
-        def skip(*args, **kwargs):
-            pass
-
-        # Skip the initializer step. This accelerates the loading
-        # of the models, especially for the quantized models
-        torch.nn.init.kaiming_uniform_ = skip
-        torch.nn.init.uniform_ = skip
-        torch.nn.init.normal_ = skip
-
-        pretrained_model_name_or_path = (
-            pretrained_model_name_or_path.as_posix()
-            if isinstance(pretrained_model_name_or_path, Path)
-            else pretrained_model_name_or_path
-        )
-
-        pretrained_model_name_or_path = download_model_directory(
-            pretrained_model_name_or_path, **kwargs
-        )
-
-        # instantiate compressor from model config
-        compressor = ModelCompressor.from_pretrained(
-            pretrained_model_name_or_path, **kwargs
+class SparseAutoModelForCausalLM:
```

Comment:
As discussed previously, please make this either a subclass with a depreciation warning, or a class which raises an error